### PR TITLE
Use Python 3.8

### LINKED ISSUE #88: Use Python 3.8 in docker images
Since both `dask` and `distributed` now support Python 3.8, should we bump our docker images here to use Python 3.8 too?

### PR DESCRIPTION
**Pull request by jrbourbeau**
Closes #88

### CODE REVIEW

**Comment by jrbourbeau** [comment]
Unfortunately it looks like Python 3.7 is still being used for the `notebook` build (note "py37" in https://travis-ci.com/github/dask/dask-docker/builds/153283500#L1329).

The `jupyter/base-notebook` Dockerfile specifies a `PYTHON_VERSION` build argument which can be used to specify the Python version that is installed and pinned:

https://github.com/jupyter/docker-stacks/blob/b782013403699cb876ed763f85c7c17450a41474/base-notebook/Dockerfile#L89-L90

I've been able to use `PYTHON_VERSION` successfully when building just `jupyter/base-notebook` locally, but it doesn't seem to work when building our `notebook` Dockerfile which uses `FROM jupyter/base-notebook`. Still looking into why `ARG PYTHON_VERSION=3.8` isn't doing the trick

File: notebook/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+ARG PYTHON_VERSION=3.8
```
Comment (jacobtomlinson) [inline notebook/Dockerfile]: Should this be `ENV`?

Comment (seanfarley) [reply notebook/Dockerfile]: Gentle ping to change this to `ENV` :-)

Comment (jrbourbeau) [reply notebook/Dockerfile]: Thanks for the ping @seanfarley

Hrm I think `ARG` is the only instruction that can precede a `FROM`. If I update this to be `ENV PYTHON_VERSION=3.8`, then running `docker-compose build notebook` locally gives me:

```
Building notebook
ERROR: No build stage in current context
```

Comment (jrbourbeau) [reply notebook/Dockerfile]: I asked about this over in the Jupyter community discourse https://discourse.jupyter.org/t/specify-python-version-when-base-notebook-is-used-as-a-base-image/3732

**Comment by martindurant** [comment]
What's the status here? I don't suppose there's any rush, but since work has been done, it would be good to include or discard.

**Comment by martindurant** [comment]
ping @jrbourbeau

**Comment by seanfarley** [comment]
I believe this is still being blocked by https://discourse.jupyter.org/t/specify-python-version-when-base-notebook-is-used-as-a-base-image/3732. Perhaps someone can ping a jupyter dev?

**Comment by jacobtomlinson** [comment]
Looking at this again I don't think `ARG` works this way.

Arguments allow you to pass variables to a Dockerfile at build time. So in this instance it allows you to build `jupyter/base-notebook` with an arbitrary Python version.

However these values cannot be inherited. This is because when we build `dask/notebook` we are pulling `jupyter/base-notebook` from Docker Hub which has already been built (probably with a default Python version).

I think what we need to do here is to first build out own local copy of `jupyter/base-notebook` specifying `3.8` as a build arg. Then build our Dask notebook image on top of that.

I'll push some commits onto here to try this out.

**Comment by jacobtomlinson** [comment]
Looks like this change has also bumped Jupyter Lab to v2. This has breaking implications for the Dask Labextension so that will need bumping too. Probably not a bad thing but is creeping the cope of this PR.

**Comment by jacobtomlinson** [comment]
[Looks like that did it.](https://travis-ci.com/github/dask/dask-docker/builds/167124311#L2245)

cc @jrbourbeau for review.

**Comment by jrbourbeau** [comment]
This looks great, thanks for pushing this over the finish line @jacobtomlinson!

**Pull request merged**